doc/source/configuration/kayobe-automation.rst

```diff
@@ -0,0 +1,126 @@
+    Once the key has been obtained, add it to :code:`secrets.yml` under :code:`secrets_github_access_token`
+
+7. Run :code:`kayobe playbook run ${KAYOBE_CONFIG_PATH}/ansible/deploy-github-runner.yml`
+
+8. Check runners have registered properly by visiting the repository's :code:`Action` tab -> :code:`Runners` -> :code:`Self-hosted runners`
+
+9. Repeat the above steps for each environment you intend to deploy runners within.
+    You can share the fine-grain access token between environments.
+
+Workflow Deployment
+-------------------
+
+1. Edit `inventory/group_vars/github-writer/writer.yml` in the base configuration making the appropriate changes to your deployments specific needs see documentation for `stackhpc.kayobe_workflows.github <https://github.com/stackhpc/ansible-collection-kayobe-workflows/tree/main/roles/github>`__
+
+2. Run :code:`kayobe playbook run ${KAYOBE_CONFIG_PATH}/ansible/write-workflows.yml`
+
+3. Add all required secrets to repository either via the GitHub UI or GitHub CLI (may require repository owner)
+    * KAYOBE_AUTOMATION_SSH_PRIVATE_KEY: private key used by Ansible to authenticate with machines.
+    * KAYOBE_VAULT_PASSWORD: password used by the config to encrypt Ansible Vault secrets.
+    * REGISTRY_PASSWORD: password used to login to the docker registry such as Pulp.
+    * TEMPEST_OPENRC: contents of :code:`kolla/public-openrc.sh`
+
+Note if you are using multiple environments and not sharing secrets between environments then each of must have the environment name prefix for each enviromment for example
+    * PRODUCTION_KAYOBE_AUTOMATION_SSH_PRIVATE_KEY
+    * PRODUCTION_KAYOBE_VAULT_PASSWORD
+    * PRODUCTION_REGISTRY_PASSWORD
+    * PRODUCTION_TEMPEST_OPENRC
+    * STAGING_KAYOBE_AUTOMATION_SSH_PRIVATE_KEY
+    * STAGING_KAYOBE_VAULT_PASSWORD
+    * STAGING_REGISTRY_PASSWORD
+    * STAGING_TEMPEST_OPENRC
+
+4. Commit and push all newly generated workflows found under :code:`.github/workflows`
+
+Final Steps
+-----------
+
+Some final steps include the following: running config-diff will require that :code:`.automation.conf/config.sh` contains a list :code:`KAYOBE_CONFIG_VAULTED_FILES_PATHS_EXTRA` of all vaulted files contained within the config.
+All such files can be found with :code:`grep -r "$ANSIBLE_VAULT;1.1;AES256" .` though make sure NOT to include `kolla/passwords.yml` and `secrets.yml`
+Also make sure tempest has been configured appropriately in :code:`.automation.conf/config.sh` to meet the limitations of a given deployment such as not using a too high of :code:`TEMPEST_CONCURRENCY` value and that overrides and load/skips lists are correct.
+Finally, once all the workflows and configuration has been pushed and reviewed you can build a kayobe image using the `Build Kayobe Docker Image` workflow. Once it is successfully built and pushed to a container registry, other workflows can be used.
```

Comment:
This could use more info on when to rebuild the Kayobe image.

Reply:
Added some information on why you might want to rebuild the image.